src/commands/fun/leaderboard.js

```diff
@@ -7,39 +7,46 @@ const Balance = require("../../schemas/balance");
 module.exports = {
     data: new SlashCommandBuilder()
         .setName("leaderboard")
-        .setDescription("Gives the amount of points a user has!"),
+        .setDescription("Displays the top 10 users with wizard points!"),
     async execute(interaction) {
         Balance.find()
             .sort({ balance: -1 })
-            .limit(10)
+            .limit(20)
             .then(async (leaderboard) => {
                 const embed = new EmbedBuilder()
                     .setColor(0x0f4a00)
                     .setTitle("Wizard Point Leaderboard");
-                let position = 1;
+                let position = 0; //Whenever a valid user is found, position is +1 and used in embed
+                let counter = 0; //Whenever any user is found, the counter for the loop goes up by one
 
-                for (record of leaderboard) {
+
+                //counter < leaderboard.length prevents reading past the number of entrees that exist
+                //position <= 10 caps the leaderboard at 10 spots
+
+                while (position < 10 && counter < leaderboard.length) { //counter < leaderboard.length prevents reading past the number of entrees that exist
+                    let record = leaderboard[counter] //The users information inside the database is stored in variable 'record'
                     const member = await interaction.guild.members
-                        .fetch(record.userId)
+                        .fetch(record.userId) //.fetch grabs the information userId stored inside record to use in output
```

Comment:
Just to clarify-
`interaction.guild.members.fetch(record.userId)` gets the information on a guild member via the Discord gateway. `record.userId` is simply their user ID, as can be seen with developer mode enabled on Discord.